src/middleware/auth.ts

```diff
@@ -0,0 +1,41 @@
+import * as jwt from 'jsonwebtoken';
+import 'dotenv/config';
+import { Request, Response, NextFunction } from 'express';
+import { CurrentUser } from '../types';
+
+export async function verifyToken(req: Request, res: Response, next: NextFunction) {
+  const authHeader = req.headers.authorization;
+
+  if(!authHeader) {
+    return res.status(401).send({
+      "data": null,
+      "error": {
+        "message": "User is not authorized"
+      }
+    });
+  }
+
+  const [tokenType, token] = authHeader.split(' ');
+
+  if(tokenType !== 'Bearer') {
+    return res.status(401).send({
+      "data": null,
+      "error": {
+        "message": "User is not authorized"
+      }
+    });
+  }
+
+  try {
+    const user = jwt.verify(token, process.env.TOKEN_KEY!) as CurrentUser;
+    req.user = user;
```

Comment:
S: You should not rely on the user data provided on the token and instead you should get the user from the source of truth (your database). This is important because in case the user is disabled or deleted from the database they must not be able to access the resources.